internal/agent/config/config.go

```diff
@@ -150,6 +154,12 @@ type TPM struct {
 	StorageFilePath string `json:"storage-file-path,omitempty"`
 }
 
+type Pruning struct {
+	// Enabled controls whether automatic pruning is enabled.
+	// Default: false
+	Enabled bool `json:"enabled,omitempty"`
```

Comment:
what other configurations do we expect here in the future that would justify a root level key vs pruneEnabled. pruning is also somewhat vauge should this be more descriptive?